src/BackupServiceProvider.php

```diff
@@ -49,6 +52,7 @@ public function packageRegistered(): void
         $this->app->singleton(ConsoleOutput::class);
 
         $this->app->bind(CleanupStrategy::class, config('backup.cleanup.strategy'));
+        $this->app->bind(TemporaryDirectoryContract::class, fn() => new TemporaryDirectoryAdapter(config('backup.temporary_directory') ?? storage_path('app/backup-temp')));
```

Comment:
Added a contract and adapter feels a bit like overkill to me. Maybe just bind the `backup-temporary-project` string to a function that returns a new instance of temporary directory directly.

In your project you could just change the binding to whatever you want.

Reply:
Thanks for the feedback. I have removed both the contract and the adapter and just bound the implementation to a string.